src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -325,6 +325,175 @@ std::optional<Values> Visitor::visit(Parser::ValuesClauseContext* ctx) {
+  // given pattern and then deletes the matched triples. This is a shortcut for
+  // a modify query.
+  reportNotSupported(ctx, "SPARQL 1.1 Update DeleteWhere is not supported.");
+}
+
+// ____________________________________________________________________________________
+ParsedQuery Visitor::visit(Parser::ModifyContext* ctx) {
+  if (ctx->iri()) {
+    // Could also be default; disallow completely for now.
+    reportNotSupported(ctx->iri(), "Named graphs are not supported.");
+  }
+  if (!ctx->usingClause().empty()) {
+    reportNotSupported(ctx->usingClause(0),
+                       "Using is not supported in Update/Insert.");
+  }
+
+  parsedQuery_._rootGraphPattern = visit(ctx->groupGraphPattern());
+
+  parsedQuery_._clause = parsedQuery::UpdateClause();
+  if (ctx->deleteClause()) {
+    parsedQuery_.updateClause().toDelete_ = visit(ctx->deleteClause());
+  }
+  if (ctx->insertClause()) {
+    parsedQuery_.updateClause().toInsert_ = visit(ctx->insertClause());
+  }
+
+  return parsedQuery_;
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::DeleteClauseContext* ctx) {
+  return visit(ctx->quadPattern());
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::InsertClauseContext* ctx) {
+  return visit(ctx->quadPattern());
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::QuadPatternContext* ctx) {
+  return visit(ctx->quads());
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::QuadDataContext* ctx) {
+  auto quads = visit(ctx->quads());
+  auto checkAndReportVar = [&ctx](const TripleComponent& term) {
+    if (term.isVariable()) {
+      reportError(ctx->quads(), "Variables (" + term.getVariable().name() +
+                                    ") are not allowed here.");
+    }
+  };
+
+  for (const auto& quad : quads) {
+    checkAndReportVar(quad.s_);
+    checkAndReportVar(quad.p_);
+    checkAndReportVar(quad.o_);
+  }
+
+  return quads;
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::QuadsContext* ctx) {
+  if (!ctx->quadsNotTriples().empty()) {
+    // Could also be default; disallow completely for now.
+    reportNotSupported(ctx->quadsNotTriples(0),
+                       "Named graphs are not supported.");
+  }
+
+  AD_CORRECTNESS_CHECK(ctx->triplesTemplate().size() == 1);
+
+  auto convertAndRegisterTriple =
+      [this](const std::array<GraphTerm, 3>& triple) -> SparqlTripleSimple {
+    registerIfVariable(triple[0]);
+    registerIfVariable(triple[1]);
+    registerIfVariable(triple[2]);
```

Comment:
If this rule doesn't appear in the WHERE clause but only in the update Header, then it is possibly wrong to call `registerIfVariable` here. This is for variables in the query body, that are visible.

Reply:
It depends. In `deleteClause` and `insertClause` they are not in the body. In `deleteWhere` they are once in the body and once in the header. But as I left out `deleteWhere` they could currently not be in the body.